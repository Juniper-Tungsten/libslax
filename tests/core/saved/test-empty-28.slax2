version 1.1;

ns bit extension = "http://xml.libslax.org/bit";

var $a = 63;
var $b = {
    expr "10111";
}

match / {
    <out> {
        <bit-set> {
            <a1> bit:set("10000", 0);
            <a2> bit:set("10000", 3);
            <a3> bit:set("10000", 5);
            <a4> bit:set("10000", 8);
        }
        <bit-clear> {
            <a1> bit:clear("11111", 0);
            <a2> bit:clear("11111", 3);
            <a3> bit:clear("11111", 5);
            <a4> bit:clear("11111", 8);
        }
        <bit-compare> {
            <a1> bit:compare(13, 14);
            <a2> bit:compare(14, 12);
            <a3> bit:compare(16, 15);
            <a4> bit:compare("10000", 16);
            <a5> bit:compare("11111", "10000");
            <a6> bit:compare("10101", "101010");
            <a7> bit:compare("101010", "10101");
        }
        <bit-and> {
            <a1> bit:and("10", "01");
            <a2> bit:and("101100", "100101");
            <a3> bit:and("0110", "1010101");
            <a4> bit:and("11111111", "1111");
            <a5> bit:and("111", "1111111");
            <a6> bit:and("101", "1111111");
            <a7> bit:and("10111", "1111101");
            <a8> bit:and(256 + 8, 256 + 8 + 1);
            <a9> bit:and($a, $b);
            <a10> bit:and($a, number($b));
        }
        <bit-or> {
            <a1> bit:or("10", "01");
            <a2> bit:or("101100", "100101");
            <a3> bit:or("0110", "1010101");
            <a4> bit:or("11111111", "1111");
            <a5> bit:or("111", "1111111");
            <a6> bit:or("101", "1111111");
            <a7> bit:or("10111", "1111101");
            <a8> bit:or(256 + 8, 256 + 8 + 1);
            <a9> bit:or($a, $b);
            <a10> bit:or($a, number($b));
        }
        <bit-mask> {
            <a1> bit:mask(0);
            <a2> bit:mask(1);
            <a3> bit:mask(4);
            <a4> bit:mask(4, 8);
            <a5> bit:mask(8, 32);
            <a6> bit:mask(10, 8);
            <a7> bit:mask(3, 3);
            <a8> bit:mask(7, 3);
            <a9> bit:mask(3, 7);
            <a10> bit:not(bit:mask(3, 7));
        }
        <ops> {
            <a1> bit:to-int("10101");
            <a2> bit:from-int(65535);
            <a3> bit:from-int(15, 10);
            <a4> bit:to-hex("10101");
            <a5> bit:from-hex("0x345");
        }
    }
}
