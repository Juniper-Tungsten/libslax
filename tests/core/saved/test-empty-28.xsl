<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" xmlns:bit="http://xml.libslax.org/bit" version="1.0" extension-element-prefixes="bit">
  <xsl:variable name="a" select="63"/>
  <xsl:variable name="b">
    <xsl:text>10111</xsl:text>
  </xsl:variable>
  <xsl:template match="/">
    <out>
      <bit-set>
        <a1>
          <xsl:value-of select="bit:set(&quot;10000&quot;, 0)"/>
        </a1>
        <a2>
          <xsl:value-of select="bit:set(&quot;10000&quot;, 3)"/>
        </a2>
        <a3>
          <xsl:value-of select="bit:set(&quot;10000&quot;, 5)"/>
        </a3>
        <a4>
          <xsl:value-of select="bit:set(&quot;10000&quot;, 8)"/>
        </a4>
      </bit-set>
      <bit-clear>
        <a1>
          <xsl:value-of select="bit:clear(&quot;11111&quot;, 0)"/>
        </a1>
        <a2>
          <xsl:value-of select="bit:clear(&quot;11111&quot;, 3)"/>
        </a2>
        <a3>
          <xsl:value-of select="bit:clear(&quot;11111&quot;, 5)"/>
        </a3>
        <a4>
          <xsl:value-of select="bit:clear(&quot;11111&quot;, 8)"/>
        </a4>
      </bit-clear>
      <bit-compare>
        <a1>
          <xsl:value-of select="bit:compare(13, 14)"/>
        </a1>
        <a2>
          <xsl:value-of select="bit:compare(14, 12)"/>
        </a2>
        <a3>
          <xsl:value-of select="bit:compare(16, 15)"/>
        </a3>
        <a4>
          <xsl:value-of select="bit:compare(&quot;10000&quot;, 16)"/>
        </a4>
        <a5>
          <xsl:value-of select="bit:compare(&quot;11111&quot;, &quot;10000&quot;)"/>
        </a5>
        <a6>
          <xsl:value-of select="bit:compare(&quot;10101&quot;, &quot;101010&quot;)"/>
        </a6>
        <a7>
          <xsl:value-of select="bit:compare(&quot;101010&quot;, &quot;10101&quot;)"/>
        </a7>
      </bit-compare>
      <bit-and>
        <a1>
          <xsl:value-of select="bit:and(&quot;10&quot;, &quot;01&quot;)"/>
        </a1>
        <a2>
          <xsl:value-of select="bit:and(&quot;101100&quot;, &quot;100101&quot;)"/>
        </a2>
        <a3>
          <xsl:value-of select="bit:and(&quot;0110&quot;, &quot;1010101&quot;)"/>
        </a3>
        <a4>
          <xsl:value-of select="bit:and(&quot;11111111&quot;, &quot;1111&quot;)"/>
        </a4>
        <a5>
          <xsl:value-of select="bit:and(&quot;111&quot;, &quot;1111111&quot;)"/>
        </a5>
        <a6>
          <xsl:value-of select="bit:and(&quot;101&quot;, &quot;1111111&quot;)"/>
        </a6>
        <a7>
          <xsl:value-of select="bit:and(&quot;10111&quot;, &quot;1111101&quot;)"/>
        </a7>
        <a8>
          <xsl:value-of select="bit:and(256 + 8, 256 + 8 + 1)"/>
        </a8>
        <a9>
          <xsl:value-of select="bit:and($a, $b)"/>
        </a9>
        <a10>
          <xsl:value-of select="bit:and($a, number($b))"/>
        </a10>
      </bit-and>
      <bit-or>
        <a1>
          <xsl:value-of select="bit:or(&quot;10&quot;, &quot;01&quot;)"/>
        </a1>
        <a2>
          <xsl:value-of select="bit:or(&quot;101100&quot;, &quot;100101&quot;)"/>
        </a2>
        <a3>
          <xsl:value-of select="bit:or(&quot;0110&quot;, &quot;1010101&quot;)"/>
        </a3>
        <a4>
          <xsl:value-of select="bit:or(&quot;11111111&quot;, &quot;1111&quot;)"/>
        </a4>
        <a5>
          <xsl:value-of select="bit:or(&quot;111&quot;, &quot;1111111&quot;)"/>
        </a5>
        <a6>
          <xsl:value-of select="bit:or(&quot;101&quot;, &quot;1111111&quot;)"/>
        </a6>
        <a7>
          <xsl:value-of select="bit:or(&quot;10111&quot;, &quot;1111101&quot;)"/>
        </a7>
        <a8>
          <xsl:value-of select="bit:or(256 + 8, 256 + 8 + 1)"/>
        </a8>
        <a9>
          <xsl:value-of select="bit:or($a, $b)"/>
        </a9>
        <a10>
          <xsl:value-of select="bit:or($a, number($b))"/>
        </a10>
      </bit-or>
      <bit-mask>
        <a1>
          <xsl:value-of select="bit:mask(0)"/>
        </a1>
        <a2>
          <xsl:value-of select="bit:mask(1)"/>
        </a2>
        <a3>
          <xsl:value-of select="bit:mask(4)"/>
        </a3>
        <a4>
          <xsl:value-of select="bit:mask(4, 8)"/>
        </a4>
        <a5>
          <xsl:value-of select="bit:mask(8, 32)"/>
        </a5>
        <a6>
          <xsl:value-of select="bit:mask(10, 8)"/>
        </a6>
        <a7>
          <xsl:value-of select="bit:mask(3, 3)"/>
        </a7>
        <a8>
          <xsl:value-of select="bit:mask(7, 3)"/>
        </a8>
        <a9>
          <xsl:value-of select="bit:mask(3, 7)"/>
        </a9>
        <a10>
          <xsl:value-of select="bit:not(bit:mask(3, 7))"/>
        </a10>
      </bit-mask>
      <ops>
        <a1>
          <xsl:value-of select="bit:to-int(&quot;10101&quot;)"/>
        </a1>
        <a2>
          <xsl:value-of select="bit:from-int(65535)"/>
        </a2>
        <a3>
          <xsl:value-of select="bit:from-int(15, 10)"/>
        </a3>
        <a4>
          <xsl:value-of select="bit:to-hex(&quot;10101&quot;)"/>
        </a4>
        <a5>
          <xsl:value-of select="bit:from-hex(&quot;0x345&quot;)"/>
        </a5>
      </ops>
    </out>
  </xsl:template>
</xsl:stylesheet>
