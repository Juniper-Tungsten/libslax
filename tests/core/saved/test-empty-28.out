<?xml version="1.0"?>
<out>
  <bit-set>
    <a1>10001</a1>
    <a2>11000</a2>
    <a3>110000</a3>
    <a4>100010000</a4>
  </bit-set>
  <bit-clear>
    <a1>11110</a1>
    <a2>10111</a2>
    <a3>011111</a3>
    <a4>000011111</a4>
  </bit-clear>
  <bit-compare>
    <a1>-1</a1>
    <a2>1</a2>
    <a3>1</a3>
    <a4>0</a4>
    <a5>1</a5>
    <a6>-1</a6>
    <a7>1</a7>
  </bit-compare>
  <bit-and>
    <a1>00</a1>
    <a2>100100</a2>
    <a3>0000100</a3>
    <a4>00001111</a4>
    <a5>0000111</a5>
    <a6>0000101</a6>
    <a7>0010101</a7>
    <a8>100001000</a8>
    <a9>010111</a9>
    <a10>00000000111111</a10>
  </bit-and>
  <bit-or>
    <a1>11</a1>
    <a2>101101</a2>
    <a3>1010111</a3>
    <a4>11111111</a4>
    <a5>1111111</a5>
    <a6>1111111</a6>
    <a7>1111111</a7>
    <a8>100001001</a8>
    <a9>111111</a9>
    <a10>10011101111111</a10>
  </bit-or>
  <bit-mask>
    <a1>0</a1>
    <a2>1</a2>
    <a3>1111</a3>
    <a4>00001111</a4>
    <a5>00000000000000000000000011111111</a5>
    <a6>1111111111</a6>
    <a7>111</a7>
    <a8>1111111</a8>
    <a9>0000111</a9>
    <a10>1111000</a10>
  </bit-mask>
  <ops>
    <a1>21</a1>
    <a2>1111111111111111</a2>
    <a3>0000001111</a3>
    <a4>0x15</a4>
    <a5>1101000101</a5>
  </ops>
</out>
